### PR TITLE
feat(proactive,surfaces): notify-channel resolver + Slack bot-channel discovery

### PR DESCRIPTION
**Pull request by khaliqgant**
## Summary
Extracts the dynamic proactive-channel primitives that shipped in [AgentWorkforce/sage#99](https://github.com/AgentWorkforce/sage/pull/99) into `@agent-assistant/surfaces` and `@agent-assistant/proactive` so any harness consumer gets them, not just sage. Sage will follow up with a PR that adopts these and deletes its local copies.

### `@agent-assistant/surfaces`
- `listBotChannels(token, botUserId)` — `users.conversations` wrapper with cursor pagination up to 3 pages
- `BotChannel` type

### `@agent-assistant/proactive`
- `PrefStore` interface (pluggable `get`/`put`) + `get/set/incrementUnconfirmedPosts` keyed by `workspaceId`
- `ChatFn` interface for dependency-injected LLM calls
- `pickChannel` — topic-matched selection via injected classifier, deterministic single-channel fast path, safe fallback on invalid JSON / unknown channelId / throw
- `parseConfirmReply` / `parseRedirectChannelName` / `normalizeChannelName` for freeform reply parsing (`yes` / `#channel-name`)
- `resolveNotifyChannel` — precedence chain `bodyOverride → confirmed pref → unconfirmed pref (counter + auto-confirm after 3 posts) → discovery + LLM pick`

Adds `@agent-assistant/surfaces` as a dependency of `@agent-assistant/proactive` for the `BotChannel` type.

## Design notes
- Nothing is coupled to Cloudflare, OpenRouter, or sage. Consumers supply a `PrefStore`, a `ChatFn`, and a `listChannels` function.
- `PrefStore` lives in `@agent-assistant/proactive` for now — can be promoted to `@agent-assistant/core` if a second consumer wants it.
- `incrementUnconfirmedPosts` is a non-atomic read-modify-write. Under concurrent writers worst case is one extra confirm prompt before auto-confirm fires; documented inline.

## Test plan
- [x] `npm run build --workspaces --if-present` — clean across all packages
- [x] `npx vitest run` in `packages/proactive` — 86/86 (53 pre-existing, 33 new)
- [x] `npx vitest run` in `packages/surfaces` — 56/56 (52 pre-existing, 4 new)
- [ ] After merge + publish: sage follow-up PR swaps local copies for `@agent-assistant/*` imports and ships a `CloudflareKvPrefStore` adapter

## Follow-up
Sage PR to adopt: tracked via memory entry in this worktree; will open against `AgentWorkforce/sage` after this ships.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/proactive/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "typecheck": "tsc --noEmit -p tsconfig.json"
   },
   "dependencies": {
+    "@agent-assistant/surfaces": ">=0.1.0",
```
[the remark this answers is not in the document]
Comment (khaliqgant) [reply packages/proactive/package.json]: Fixed in ccacd03. Bumped both `@agent-assistant/surfaces` and `@agent-assistant/proactive` to 0.2.19 and pinned proactive's surfaces dep to `>=0.2.19` so consumers can't resolve to a surfaces release that predates `BotChannel`.

File: packages/proactive/src/notify-channel-reply.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  | { kind: 'none' };
+
+export function normalizeChannelName(channel: string): string {
+  return channel.trim().replace(/^#/, '').toLowerCase();
+}
+
+export function parseRedirectChannelName(text: string): string | undefined {
+  const trimmed = text.trim();
+  if (!trimmed.startsWith('#')) {
+    return undefined;
```
[the remark this answers is not in the document]
Comment (khaliqgant) [reply packages/proactive/src/notify-channel-reply.ts:68]: Fixed in ccacd03. `parseRedirectChannelName` and `parseConfirmReply` now accept Slack's auto-rewritten mention form `<#C123|name>` in addition to the literal `#name`. The return shape widened from `string` to `{ channelId?: string; channelName?: string }` so callers can prefer ID lookup (more robust than name lookup) and fall back to name when only a literal was typed. Added 5 tests covering both forms plus edge cases. No adopting consumer has wired to these modules yet, so the API widening is safe.
